app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -453,8 +453,9 @@ public void setLiveData(LiveData<PagedList<SearchTeiModel>> liveData) {
                     liveAdapter.submitList(data.val0());
                     binding.progressLayout.setVisibility(View.GONE);
                 } else {
-                    binding.filterCounter.setVisibility(View.GONE);
-                    binding.searchFilterGeneral.setVisibility(View.GONE);
+                    boolean filtersActive = FilterManager.getInstance().getTotalFilters() != 0;
+                    binding.filterCounter.setVisibility(filtersActive ? View.VISIBLE : View.GONE);
+                    binding.searchFilterGeneral.setVisibility(filtersActive ? View.VISIBLE : View.GONE);
```

Comment:
I know that this class is not new, but what do you think if we do this instead 

presenter.updateFilterCount(filtersActive)

SearchTePresenter.class

fun updateFilterCount(isActive:Boolean){
   if (isActive) do something
   else do this
}

Then it can be easily add to our searchTepresenterTest and reduce this debt

Let me know what you think about it

Reply:
The filterCounter already updates the value using an observableField. This is just to not hide the filter button which requires first to have the result from the search.

Reply:
Sorry I meant FilterCounter XD

fun updateFilterCounter ...

Reply:
done